mbtools/copy_resources_s3.py

```diff
@@ -0,0 +1,137 @@
+import hashlib
+import json
+import os
+import sys
+import magic
+import boto3
+import botocore
+from pathlib import Path
+
+
+def upload_resources(resource_dir, metadata_dir, bucket, s3_dir):
+    """Uplaods resources to s3 if they dont already exist there"""
+    hash_to_filename_map = new_resource_hashes(resource_dir)
+    hashes_new = list(hash_to_filename_map.keys())
+    hashes_metadata = existing_metadata_hashes(metadata_dir)
+    hashes_s3 = existing_s3_hashes(bucket, s3_dir)
+    hashes_new = compare_and_remove_hashes(hashes_new, hashes_metadata)
```

Comment:
In lieu of this `compare_and_remove_hashes` function, you might be able to do a one-liner with `filter()`:

```python
hashes_to_upload = list(filter(lambda hash: hash not in hashes_metadata, hashes_new))
```